docs/migration.md

```diff
@@ -0,0 +1,418 @@
+                break;
+            case "/pong":
+                this.messageService.send(streamId, "ping");
+                break;
+        }
+    }
+}
+
+
+@Component
+public class RealTimeEventComponent {
+
+    private final PollService pollService;
+
+    public RealTimeEventComponent(PollService pollService) {
+        this.pollService = pollService;
+    }
+
+    @EventListener
+    public void onMessageSent(RealTimeEvent<V4MessageSent> event) {
+        this.pollService.handleIncomingMessage(event.getSource().getMessage,
+                StreamType.TypeEnum.formValue(event.getSource().getMessage.getStream.getStreamType()));
+    }
+}
+```
+</td>
+</tr>
+</table>
+
+An example of no framework based project using `SymphonyBdk` to retrieve BDK services:
+````java
+@Slf4j
+public class GreetingsAllRoomsBot {
+
+  public static void main(String[] args) throws Exception {
+
+    final SymphonyBdk bdk = new SymphonyBdk(loadFromSymphonyDir("config.yaml"));
+
+    // list all rooms
+    Stream<StreamAttributes> rooms = bdk.streams().listAllStreams(new StreamFilter());
+    
+    rooms.forEach(streamAttributes -> {
+          // send message to room
+          bdk.messages().send(streamAttributes.getId(), "Hello world!");
+          log.info("Message sent to room with: id:{}, name:{}", streamAttributes.getId(),
+              streamAttributes.getRoomAttributes().getName());
+    });
+    
+    bdk.datafeed().start();
+  }
+}
+````
+> A list of BDL available services can be found [here](./fluent-api.md)
+
+## Event listeners
+Java BDK 2.0 comes with a simplified way to handle event listeners.
+<table>
+<tr>
+<th>Java SDK</th>
+<th>Java BDK 2.0</th>
+</tr>
+<tr>
+<td>
+
+In Java SDK, the bot had to implement 3 listeners classes: 
+- one for IM (1 to 1 conversation)
+- one for MIM (room)
+- one for Symphony elements
+```java
+@Slf4j
+@Service
+public class ElementsListenerImpl implements ElementsListener {
+    public void onElementsAction() {...}
+}
+
+@Slf4j
+@Service
+public class ElementsListenerImpl implements ElementsListener {
+    public void onIMMessage() {...}
+}
+
+@Slf4j
+@Service
+public class RoomListenerImpl implements RoomListener {
+    public void onRoomMessage() {...}
+}
+```
+</td>
+<td>
+
+In Java BDK 2.0, only one component `RealTimeEventComponent` has to be implemented with two methods having `@EventListener` annotation. The 3 classes can be factored in one single component. *(The example below uses a Spring Boot based project)* 
+```java
+@Component
+public class RealTimeEventComponent {
+    @Autowired
+    private MessageService messageService;
+
+    @EventListener
+    public void onMessageSent() {...}
+
+    @EventListener
+    public onElementsAction() {...}
+}
+```
+</td>
+</tr>
+</table>
+
+## Models
+Some models names have been changed in Java BDK 2.0. This requires to change some variables types names in your legacy bots.
+For example : *(non exhaustive list)*
```

Comment:
it would be better to have an exhaustive list of changed models.
Is this documented somewhere that we can link here? 
Is it useful to provide an exhaustive list of them in a migration guide?

Reply:
This is not documented as far as I know. And I am not sure this would be useful to have an exhaustive list (too much effort in comparison with the value). However, we could say that the models follow the ones in the swagger specification of the public API and field names in java classes correspond the field names in the json payload